forc/src/ops/forc_run.rs

```diff
@@ -51,27 +51,29 @@ pub async fn run(command: RunCommand) -> Result<Vec<fuel_tx::Receipt>> {
         TxParameters::new(command.byte_price, command.gas_limit, command.gas_price),
     );
 
+    let node_url = match &manifest.network {
+        Some(network) => &network.url,
+        _ => &command.node_url,
+    };
```

Comment:
I think we want the `command.node_url` to have priority here. E.g. I can imagine the user having a URL specified in their manifest, but wanting to temporarily test another by passing a flag. It also might be strange to have the flag just be ignored in the case that a URL is specified in the manifest.

Reply:
Yeah I agree. We want the flag to take priority here if one was passed in.

Reply:
@mitchmindtree @JoshuaBatty 
- I see the `command.node_url` has a `default_value`, so it will always be present. How are we handling that in this case? 
- Always just use `command.node_url` when `command.simulate_tx` is passed? 
- Or maybe use `manifest.network.url` in the case that `command.simulate_tx` is passed and `command.node_url` is the default value?

Reply:
Ahh good spot!

I think we should make `command.node_url` an `Option`, move that default URL out into a constant, and then work out the target node URL by:

1. Checking if `command.node_url` is `Some`,
2. if `None`, use `manifest.network.url`,
3. If that's also `None`, fallback to the default URL (the one that is currently specified [here](https://github.com/FuelLabs/sway/blob/98f261fcc81b14fbbc742e1974460517d11bc653/forc/src/cli/commands/run.rs#L26)).

How does that sound?